packages/oscal-react-library/src/components/OSCALMetadata.tsx

```diff
@@ -306,34 +308,72 @@ const MetadataInfoList: React.FC<MetadataInfoListProps> = (props) => {
   );
 };
 
-export interface OSCALMetadataPartyProps {
+interface OSCALMetadataPartyLogoProps extends ResolvableLinkHref {
+  link: OSCALLink;
+  partyName?: string;
+}
+
+const OSCALMetadataPartyLogo: React.FC<OSCALMetadataPartyLogoProps> = ({
+  link,
+  backMatter,
+  parentUrl,
+  partyName,
+}) => {
+  let logoUri = undefined;
+
+  if (backMatter) {
+    try {
+      logoUri = resolveLinkHref(backMatter, link.href, parentUrl, /^image\//, false);
+    } catch (err) {
+      // Silently fail on unresolved diagram resources
+    }
+  }
+
+  const imageSrc = logoUri ?? link.href;
```

Comment:
This is a lot cleaner but also has the same issue that `resolveLinkHref` requires `backMatter` to be defined. I  am unsure where to put the check that makes the most sense.

Since `resolveLinkHref` more or less provides a nice wrapper around using the `BackMatterLookup` class, I thought that it doesnt make sense for you to not have `backmatter` when calling `resolveLinkHref`.

Reply:
Just add `if (!backMatter) return undefined` to the top of `maybeResolveUri`?

Reply:
I did that, you just cant do `const maybeResolveUri: typeof resolveLinkHref`